src/merger.rs

```diff
@@ -0,0 +1,81 @@
+use cosmogony::{file_format::OutputFormat, read_zones_from_file, Zone, ZoneIndex};
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use std::path::{Path, PathBuf};
+
+#[derive(Default)]
+struct CosmogonyMerger {
+    id_offset: usize,
+}
+
+fn to_json_stream(
+    mut writer: impl std::io::Write,
+    zones: impl std::iter::Iterator<Item = Zone>,
+) -> Result<(), failure::Error> {
+    for z in zones {
+        serde_json::to_writer(&mut writer, &z)?;
+        writer.write_all(b"\n")?;
+    }
+    Ok(())
+}
+
+impl CosmogonyMerger {
+    fn merge_cosmogony(
+        &mut self,
+        files: &[PathBuf],
+        mut writer: impl std::io::Write,
+    ) -> Result<(), failure::Error> {
+        for f in files {
+            self.read_cosmogony(&f, &mut writer)?;
+        }
+        Ok(())
+    }
+
+    fn read_cosmogony(
+        &mut self,
+        file: &PathBuf,
+        writer: impl std::io::Write,
+    ) -> Result<(), failure::Error> {
+        let mut max_id = 0;
+        let zones = read_zones_from_file(file)?
+            .into_iter()
+            .filter_map(|z| z.ok())
+            .map(|mut z| {
+                z.id = self.get_updated_id(z.id);
+                max_id = std::cmp::max(max_id, z.id.index);
+                z.parent = z.parent.map(|p| self.get_updated_id(p));
+                z
+            });
+        to_json_stream(writer, zones)?;
+        // we update the id_offset, for the next file
+        self.id_offset = max_id + 1;
+        Ok(())
+    }
+
+    fn get_updated_id(&self, idx: ZoneIndex) -> ZoneIndex {
+        ZoneIndex {
+            index: idx.index + self.id_offset,
+        }
+    }
+}
+
+pub fn merge_cosmogony(files: &[PathBuf], output: &Path) -> Result<(), failure::Error> {
+    let mut merger = CosmogonyMerger::default();
+
+    let format = OutputFormat::from_filename(output)?;
+    let file = std::fs::File::create(output)?;
+    let mut stream = std::io::BufWriter::new(file);
+    match format {
+        OutputFormat::JsonGz | OutputFormat::Json => panic!(
+            "cannot read real cosmogonies, only jsonl/jsonl.gz to be able to stream the files"
```

Comment:
This error message does not sound relevant. The _output_ format is validated here, not the files that will be read.  
As far as I can see, nothing prevents from using .json or .json.gz formats as input `files`.

Reply:
hum I only added the method to write streams (`to_json_stream`) as we write the zones on the go (and I don't see a usecase where someone would want to output `json` here :thinking: 

would that be okay for you if I only correct the message?